From 3d6c5ef8e2099dd38141b854a789d76252d10b04 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>
Date: Tue, 28 Jan 2020 15:33:15 -0600
Subject: [PATCH 27/37] MLK-23258-4 dts: SATA requires PCIE_A power domain to
 be ON

commit 3b469d58b7c93c18231cdc07f92e5a5a2690e4e9 from
https://source.codeaurora.org/external/imx/linux-imx.git

SATA driver write to regisers in the PCIE_A power domain and
hence PCIE_A needs to be powered on even when ONLY SATA is enabled.

Signed-off-by: Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>
(cherry picked from commit 59664a795373b14d2396bc26822aff41110ea3cc)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi | 35 +++++++++----------
 1 file changed, 17 insertions(+), 18 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index cc30fd515263..433c2cd1d113 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -490,26 +490,25 @@
 						#address-cells = <1>;
 						#size-cells = <0>;
 
-					};
-
-					pd_serdes1: PD_HSIO_SERDES_1 {
-						reg = <SC_R_SERDES_1>;
-						#power-domain-cells = <0>;
-						power-domains =<&pd_serdes0>;
-						#address-cells = <1>;
-						#size-cells = <0>;
-
-						pd_pcie1: PD_HSIO_PCIE_B {
-								reg = <SC_R_PCIE_B>;
-								#power-domain-cells = <0>;
-								power-domains =<&pd_serdes1>;
-								#address-cells = <1>;
-								#size-cells = <0>;
+						pd_serdes1: PD_HSIO_SERDES_1 {
+							reg = <SC_R_SERDES_1>;
+							#power-domain-cells = <0>;
+							power-domains =<&pd_pcie0>;
+							#address-cells = <1>;
+							#size-cells = <0>;
 
-							pd_sata0: PD_HSIO_SATA_0 {
-									reg = <SC_R_SATA_0>;
+							pd_pcie1: PD_HSIO_PCIE_B {
+									reg = <SC_R_PCIE_B>;
 									#power-domain-cells = <0>;
-									power-domains =<&pd_pcie1>;
+									power-domains =<&pd_serdes1>;
+									#address-cells = <1>;
+									#size-cells = <0>;
+
+								pd_sata0: PD_HSIO_SATA_0 {
+										reg = <SC_R_SATA_0>;
+										#power-domain-cells = <0>;
+										power-domains =<&pd_pcie1>;
+								};
 							};
 						};
 					};
-- 
2.17.1

